sqlsimplekv/keyvalue.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sqlsimplekv
+
+import (
+	"database/sql"
+	"time"
+
+	sqldriver "database/sql/driver"
+
+	"golang.org/x/net/context"
+	errgo "gopkg.in/errgo.v1"
+
+	"github.com/juju/simplekv"
+	"github.com/lxc/lxd/shared/logger"
+)
+
+// NewStores returns a new Stores instance that uses the
+// given sql database for storage, generating SQL with the
+// given driver (currently only "postgres" is supported).
+//
+// The data will be stored in a table named "keyvalue_data".
+func NewStores(driverName string, db *sql.DB) (*Stores, error) {
```

Comment:
I'd make this `NewStore(driverName string, db *sql.DB, table string)` There is no need for quite such a complex interface then.

Reply:
done